Update icon colours to $color-mid-dark

kwm14 opened the pull request:
## Done

- Update icon colors to `$color-mid-dark` so the icon set matches

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/examples/patterns/icons/icons-light/
- See that the color of the icons look the same

## Details

- Fixes https://github.com/canonical-web-and-design/design-vanilla-framework/issues/404
- Fixes #2221 

## Screenshots
<img width="942" alt="Screenshot 2019-08-05 at 16 54 53" src="https://user-images.githubusercontent.com/17748020/62477875-c426b700-b7a1-11e9-8c16-7b9155825617.png">

webteam-app commented:
Starting demo at: https://vanilla-framework-canonical-web-and-design-pr-2446.run.demo.haus/

kwm14 commented:
## QA
- Just tested the icons here: http://0.0.0.0:8101/examples/patterns/icons/icons-dark/
- Fails on the dark background as they're meant to invert and become `$color-x-light`

<img width="937" alt="Screenshot 2019-08-05 at 16 57 59" src="https://user-images.githubusercontent.com/17748020/62478274-91c98980-b7a2-11e9-9379-d87b2028c65d.png">

- Looking at the code I need to add `fill='#{vf-url-friendly-color($color)}'`
- @anthonydillon @deadlight where would I place this inside the `background-image`? I've tried after `fill-rule` but doesn't seem to work...

kwm14 commented:
## Fixed
- Removed hardcoded hex values inside SVG
- Updated icon colors in the mixin
  - Light: `$color-mid-dark`
<img width="1406" alt="Screenshot 2019-08-06 at 11 24 30" src="https://user-images.githubusercontent.com/17748020/62532765-15d04f80-b83d-11e9-86f0-55dbd539c97f.png">

  - Dark: `$color-mid-x-light` (previously was `$color-mid-light` which was slightly too dark)
<img width="1406" alt="Screenshot 2019-08-06 at 11 24 41" src="https://user-images.githubusercontent.com/17748020/62532775-1963d680-b83d-11e9-8a75-88df1c84f589.png">

webteam-app commented:
Starting demo at: https://vanilla-framework-canonical-web-and-design-pr-2446.run.demo.haus/
